### ISSUE TITLE
Backup & restore: incl. net topology, qrexec policies

### ISSUE THREAD

**Issue by rootkovska** [QubesOS/qubes-issues#1635]

**Comment by marmarek** [comment]
Net topology is already included (if not, please give concrete example). Others - yes, should be.

**Comment by axon-qubes** [comment]
Might be nice also to include:

* `/etc/default/grub`
* `/etc/crontab`
* `/etc/crypttab`
* `/etc/fstab`

**Comment by marmarek** [comment]
@axon-qubes I'm not sure about those files - those are highly specific to particular installation, if you restore the backup on fresh installation, replacing those files most likely will break the system (at least because of different UUIDs)
Maybe `/etc/crontab` worth including.

**Comment by axon-qubes** [comment]
Right, I was mainly thinking to include them in the backup but not automatically restore. But you make a good point. Perhaps it's better to leave it up to users to `cp` them to dom0's home before backing up.

**Comment by axon-qubes** [comment]
> Maybe `/etc/crontab` worth including.

Actually, I just checked, and my `/etc/crontab` has no entries, even though I have entries which show up when I do `crontab -l` (and they are working). So, backing up `/etc/crontab` would not be very useful (at least in my case).

**Comment by marmot1791** [comment]
Since this is marked as a duplicate of the feature request I raised, I'd like to propose that backup of dom0 should include:

- /etc/qubes
- /etc/qubes-rpc

and that these files should be restored to ~/restore/etc/qubes and ~/restore/etc/qubes-rpc respectively. Later maybe other files from dom0 could be included by default or a preference setting to do so could be implemented.

I would also appreciate others' input as to what documentation should accompany this. In particular, do we expect that a user can copy ~/restore/etc/qubes and ~/restore/etc/qubes-rpc into a new install's /etc/qubes and /etc/qubes-rpc without issues? Is that true only for the same release or also across point releases (e.g. R4.0 -> R4.1).

I would like to implement this and provide some pull requests, and it seems like consensus on the solution will make them more likely to be accepted.

**Comment by andrewdavidwong** [comment]
Generalizing this issue to reflect developments in Qubes OS since it was first opened.
